### PR TITLE
Inherit top-level nodeSelector in placement, horizon, keystone template

### PR DESCRIPTION
**Pull request by olliewalsh**
Depends-On: https://github.com/openstack-k8s-operators/placement-operator/pull/266
Depends-On: https://github.com/openstack-k8s-operators/horizon-operator/pull/388
Depends-On: https://github.com/openstack-k8s-operators/keystone-operator/pull/502
Jira: [OSPRH-10734](https://issues.redhat.com//browse/OSPRH-10734)

### CODE REVIEW

File: pkg/openstack/placement.go [hunk 1]
```diff
@@ -108,9 +112,6 @@ func ReconcilePlacementAPI(ctx context.Context, instance *corev1beta1.OpenStackC
 		if placementAPI.Spec.Secret == "" {
 			placementAPI.Spec.Secret = instance.Spec.Secret
 		}
-		if placementAPI.Spec.NodeSelector == nil && instance.Spec.NodeSelector != nil {
-			placementAPI.Spec.NodeSelector = instance.Spec.NodeSelector
-		}
```
Comment (gibizer) [inline pkg/openstack/placement.go:113]: I thought we set it inside the createOrPatch to actually ensure it is updated later not just during initial created.

Comment (stuggi) [reply pkg/openstack/placement.go:113]: we apply it to the `instance.Spec.Placement.Template` which added via the DeepCopyInto within the `CreateOrPatch` a couple of lines above this. so it should be the same as having this

Comment (olliewalsh) [reply pkg/openstack/placement.go:113]: `placementAPI.Spec.NodeSelector == nil` can only be true on the first invocation

Comment (olliewalsh) [reply pkg/openstack/placement.go:113]: `DeepCopyInto` will reset it to nil though, so probably did work in this case. IIRC it wasn't correct for some services

**Comment by gibizer** [comment]
looks good, just the replace line needs to be removed

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**